fips.md

```diff
@@ -0,0 +1,80 @@
+
+## Does ReadonlyREST is fully FIPS compliant?
+
+Unfortunately not yet. Currently only SSL transport part uses FIPS compliant algorithms.
+
+## How to enable SSL FIPS compliance
+
+1. Prepare keystore and truststore in BCFKS format which is FIPS compliant. Your existing JKS or PKCS12 keystore could be easily converted to BCFKS. Process is described [in this section](#how-to-convert-jkspkcs12-keystore-files-into-bcfks). 
+> :warning: BCFKS format is supported only when FIPS mode is enabled. It won't be recognised otherwise.
+
+> :warning: When using FIPS mode using different password for specific keystore elements is not supported and `key_pass` configuration field is ignored.
+
+2. Configure readonlyrest.yml to use new keystore and truststore. You will also need to add new configuration parameter `fips_mode`. Here's an example:
+``` 
+readonlyrest:
+  fips_mode: SSL_ONLY
+  ssl:
+    enable: true
+    keystore_file: "keystore.bcfks"
+    keystore_pass: readonlyrest
+    truststore_file: "truststore.bcfks"
+    truststore_pass: readonlyrest
+
+  ssl_internode:
+    enable: true
+    keystore_file: "keystore.bcfks"
+    keystore_pass: readonlyrest
+    key_pass: readonlyrest
+    truststore_file: "truststore.bcfks"
+    truststore_pass: readonlyrest
+```
+3. In case you are using ES >= 7.10 you need to modify `$JAVA_HOME/conf/security/java.policy` file and add this section at the end of it. It is required because ES restricted permissions that could be used by plugins.
+```
+grant {
+  permission org.bouncycastle.crypto.CryptoServicesPermission "exportSecretKey";
+  permission org.bouncycastle.crypto.CryptoServicesPermission "exportPrivateKey";
+  permission java.security.SecurityPermission "getProperty.jdk.tls.disabledAlgorithms";
+  permission java.security.SecurityPermission "getProperty.jdk.certpath.disabledAlgorithms";
+  permission java.security.SecurityPermission "getProperty.keystore.type.compat";
+  permission java.security.SecurityPermission "removeProvider.SunRsaSign";
+  permission java.security.SecurityPermission "removeProvider.SunJSSE";
+  permission java.io.FilePermission "${java.home}/lib/security/jssecacerts", "read";
+  permission java.io.FilePermission "${java.home}/lib/security/cacerts", "read";
+  permission java.security.SecurityPermission "getProperty.jdk.tls.server.defaultDHEParameters";
+  permission org.bouncycastle.crypto.CryptoServicesPermission "defaultRandomConfig";
+};
+```
+
+
+## How to convert JKS/PKCS12 keystore files into BCFKS
+
+1. Download [jar with bc-fips](https://repo1.maven.org/maven2/org/bouncycastle/bc-fips/1.0.2.3/bc-fips-1.0.2.3.jar) library and place it preferably in the same directory where you store keystore files to convert. 
+
+2. Open terminal and go to directory with keystore to convert
+
+3. Use keytool with following parameters to perform the conversion:
+  ```
+  keytool \
+  -importkeystore \
+  -srckeystore < filename of keystore that you want to convert >  \
```

Comment:
what do you think about changing this `< filename of keystore that you want to convert >` convention to sth like that: `KEYSTORE_FILENAME` and underneath:

```
where:
* KEYSTORE_FILENAME - filename of keystore that you want to convert 
```
?